domains/transfer/Transfer.API/API/Claiming/Automation/ClaimWorker.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+namespace API.Claiming.Automation;
+
+public class ClaimWorker : BackgroundService
+{
+    private readonly IServiceProvider serviceProvider;
+
+    public ClaimWorker(IServiceProvider serviceProvider)
+    {
+        this.serviceProvider = serviceProvider;
+    }
+
+    protected override async Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        using var scope = serviceProvider.CreateScope();
+        var claimService = scope.ServiceProvider.GetRequiredService<IClaimService>();
+
+        await claimService.Run(stoppingToken);
```

Comment:
When resolving this way, there is only created one instance of IClaimService that lives the entire application lifetime. Following the line of dependencies for ClaimService, we hit a DbContext; this instance of DbContext is never disposed and is against the recommendation. See https://learn.microsoft.com/en-us/ef/core/dbcontext-configuration/#the-dbcontext-lifetime.
You should use IDbContextFactory instead to create an instance of the DbContext when needed (which is only on the start of each run for claim automation).

Reply:
I have corrected this. Let me know if I did it wrong